Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.h

```diff
@@ -128,23 +128,30 @@ class GdbServer {
     uint8_t Data[10];
   };
 
-  struct FEX_PACKED GDBContextDefinition {
-    uint64_t gregs[FEXCore::Core::CPUState::NUM_GPRS];
-    uint64_t rip;
+  template<size_t GPRCount, typename GPRType, size_t XMMCount, size_t XMMArray>
+  struct FEX_PACKED GDBContextDefinitionBase {
+    GPRType gregs[GPRCount];
+    GPRType rip;
```

Comment:
Making all this compile-time-based increases the code complexity here a lot. Have you considered using a runtime-based approach via `std::span` instead?

```cpp
bool is_64_bit;
GPRType gregs_storage[NUM_GPRS];
std::span gregs(gregs_storage, is_64_bit ? NUM_GPRS : NUM_GPRS / 2);
```

AFAICS we only need binary serialization in one place. We can do a little more work manually there, at the benefit of saving all the template ugliness this change currently adds (most notably `GenerateAndFetchContextDefinition.template operator()<GDBContextDefinition>`...).

Reply:
I don't see what you are intending here? We give the connected gdb instance a layout that is defined via `BuildTargetXML` and we need to ensure that the resulting binary that we encode and send over the pipe matches that exact struct layout, which differs between x86-64 and i386.

This is probably just me being daft since I don't quite understand what the intended change does.

Reply:
The struct definition is technically used in both `CommandReadRegisters` and `CommandReadReg` in different ways.
But I've removed any amount of being "clever".